wallet/src/de/schildbach/wallet/ui/WalletActivity.java

```diff
@@ -377,6 +431,49 @@ public void handleRestoreWallet() {
                     REQUEST_CODE_RESTORE_WALLET);
     }
 
+    public void restoreWalletFromMnemonic() {
+        final View view = LayoutInflater.from(getApplicationContext()).inflate(R.layout.restore_wallet_mnemonic_dialog, null);
+        final EditText input = (EditText) view.findViewById(R.id.import_mnemonic);
+        final DialogBuilder builder = new DialogBuilder(WalletActivity.this);
+        builder.setTitle(R.string.import_mnemonic);
+        builder.setView(view);
+        builder.setPositiveButton(R.string.button_ok, null); // dummy, just to make it show
+        builder.setNegativeButton(R.string.button_cancel, null);
+        builder.setCancelable(false);
+        final AlertDialog dialog = builder.create();
+        dialog.setOnShowListener(new DialogInterface.OnShowListener() {
+            @Override
+            public void onShow(final DialogInterface d) {
+                final Button positiveButton = dialog.getButton(DialogInterface.BUTTON_POSITIVE);
+                positiveButton.setTypeface(Typeface.DEFAULT_BOLD);
+                positiveButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(final View v) {
+                        final List<String> codes = Arrays.asList(input.getText().toString().split("\\s+"));
+                        final DeterministicSeed seed = new DeterministicSeed(codes, null, "", wallet.getEarliestKeyCreationTime());
```

Comment:
The wallet birthdate seems wrong. You can't copy it from the previous wallet.

Reply:
Then I need to scan the entire blockchain, providing the genesis block's timestamp?

Reply:
Yes, or as @prusnak pointed out you could use Jan 2014 as there were no BIP39 wallets before.

Reply:
Actually we have constant for this date: `MnemonicCode.BIP39_STANDARDISATION_TIME_SECS`

Reply:
Awesome